src/sbvr-api/sbvr-utils.ts

```diff
@@ -1153,11 +1157,101 @@ const $getAffectedIds = async ({
 	return result.rows.map((row) => row[idField]);
 };
 
+const validateBatch = (req: Express.Request) => {
+	const { requests } = req.body as { requests: uriParser.UnparsedRequest[] };
+	if (!Array.isArray(requests)) {
+		throw new BadRequestError(
+			'Batch requests must include an array of requests in the body via the "requests" property',
+		);
+	}
+	if (req.headers != null && req.headers['content-type'] == null) {
+		throw new BadRequestError(
+			'Headers in a batch request must include a "content-type" header if they are provided',
+		);
+	}
+	if (
+		requests.find(
+			(request) =>
+				request.headers?.authorization != null ||
+				request.url?.includes('apikey='),
+		) != null
+	) {
+		throw new BadRequestError(
+			'Authorization may only be passed to the main batch request',
+		);
+	}
+	const ids = new Set<string>(
+		requests
+			.map((request) => request.id)
+			.filter((id) => typeof id === 'string') as string[],
+	);
+	if (ids.size !== requests.length) {
+		throw new BadRequestError(
+			'All requests in a batch request must have unique string ids',
+		);
+	}
+
+	for (const request of requests) {
+		if (
+			request.headers != null &&
+			request.headers['content-type'] == null &&
+			(req.headers == null || req.headers['content-type'] == null)
+		) {
+			throw new BadRequestError(
+				'Requests of a batch request that have headers must include a "content-type" header',
+			);
+		}
+		if (request.method == null) {
+			throw new BadRequestError(
+				'Requests of a batch request must have a "method"',
+			);
+		}
+		const upperCaseMethod = request.method.toUpperCase();
+		if (!validBatchMethods.has(upperCaseMethod)) {
+			throw new BadRequestError(
+				`Requests of a batch request must have a method matching one of the following: ${Array.from(
+					validBatchMethods,
+				).join(', ')}`,
+			);
+		}
+		if (
+			request.body !== undefined &&
+			(upperCaseMethod === 'GET' || upperCaseMethod === 'DELETE')
+		) {
+			throw new BadRequestError(
+				'GET and DELETE requests of a batch request must not have a body',
+			);
+		}
+	}
+
+	const urls = new Set<string | undefined>(
+		requests.map((request) => request.url),
+	);
+	if (urls.has(undefined)) {
+		throw new BadRequestError('Requests of a batch request must have a "url"');
+	}
+	if (urls.has('/university/$batch')) {
+		throw new BadRequestError('Batch requests cannot contain batch requests');
+	}
```

Comment:
Debugging or should it be? :
```suggestion
	if (urls.has('$batch')) {
		throw new BadRequestError('Batch requests cannot contain batch requests');
	}
```

Reply:
Don't remember but I suspect it's neither, just a mistake 🤦